mlir/utils/jenkins/Jenkinsfile

```diff
@@ -793,5 +812,72 @@ pipeline {
                 }
             }
         }
+        stage ("MIGraphX") {
+            when {
+                beforeAgent true;
+                equals expected: true, actual: params.checkMIGraphX;
+            }
+            matrix {
+                axes {
+                    axis {
+                        name 'CODEPATH'
+                        values 'mfma', 'vanilla', 'navi21'
+                    }
+                }
+                agent {
+                    docker {
+                        image dockerImage()
+                        args dockerArgs()
+                        label getLabelFromCodepath("${CODEPATH}")
+                        alwaysPull true
+                    }
+                }
+                when {
+                    beforeAgent true
+                    expression { return shouldRunFromCodepath("$CODEPATH") }
+                }
+                environment {
+                    HOME="${WORKSPACE}"
+                }
+                stages {
+                    stage('Environment') {
+                        steps {
+                            echo "codepath is ${CODEPATH}"
+                            showEnv()
+                        }
+                    }
+                    stage("Install MIGraphX Dependencies") {
+                        steps {
+                            sh 'apt-get install miopen-hip'
+                            sh 'cget -p /MIGraphXDeps install ${WORKSPACE} -DBUILD_MIXR_TARGET=On -DLLVM_ENABLE_ZSTD=Off -DLLVM_ENABLE_THREADS=Off'
```

Comment:
Let's not install into `/`, but go into `$WORKSPACE/MIGraphXDeps`

Reply:
well the dependencies are already there in the container. Similiar to MIOpenDeps.

Reply:
So it needs to be a location in the container which already contains the static dependencies installed.

Reply:
MIGraphX uses more than rocmlir-lib exposes mixr CAPI s and whatnot. I never got that working. It uses more MLIR headers. From what Ive heard from @jungpark-mlir, this is the only way to make it work. If someone knows how to do that Im all ears..

Reply:
And fourth, use MIGraphX build flags to turn off, for example, the MIOpen dependcy, if you can.

Reply:
> First off, any apt install steps should be in the Dockerfile.

I think you are referring miopen-hip installation.
Yes, we can, I just kept it here to avoid any installation conflicts with whatever miopen installation happens before...
if that's not an issue I can move that.

> Second, whatever build flags are needed for the MIGraphX integration - I think it's to do with headers - you can just pass to our CMake, and then use our install into a prefix.

I can have a go tomorrow and see if this works.

> Third, this way doesn't actually test against whatever the CI pipeline is being run against, it tests against MiGraphX's commit, which is a useless test for us.

This is not correct (I think). cget uses the current folder (current rocMLIR checked out version) checked out in the workspace to package. -- may I ask how did you figure out it uses MiGraphX's commit or its relevance to what rocMLIR commit's being used.

> And fourth, use MIGraphX build flags to turn off, for example, the MIOpen dependcy, if you can.

I don't this is possible MIGraphX only offloads MLIR if there are fused kernels and standalone go to MIOpen -- so its not binary in that sense.

Reply:
The other thing here is that, even though we're running in a Docker container, we want all the things we install to be confined to $WORKSPACE. So that MIGraphX build will need a CMAKE_PREFIX_PATH that has two paths - one for the global dependencies, one for the MRIR we just installed.

... Come to think of it, in the Dockerfile, why install to /MIGraphXDeps at all, just go to /usr/local the same way the MIOpen dependencies do.

Given the flags you passed to `cget`, I expected it to be running in the MIGraphX context on initial reading, and so therefore to use MIGraphX's requirements files to get and build a MLIR. Heck, it has `get` in the name. But it's quite possible that this runs on $WORKSPACE and so is therefore fine.

And so that being said, use an install target of `${WORKSPACE}/MIGraphXDeps` not `/MIGraphXDeps`

Reply:
Also, given the above, I'd like to see a very simple smoke test that runs one fused kernel through MLIR and verifies it so that we can disable out as much of MIGraphX (like that MIOpen dependency) as possible. I want this CI run to basically be "is the MIGraphX integration fundamentally broken (things don't compile/a simple example doesn't run) if you move to MLIR tip"

Reply:
Only concern with /usr/local is a potential dependency collision with other clients. It felt compartmentalized to install it to a separate folder. I still not getting the value of having to use under WORKSPACE dir. It is just installing the rocMLIR dependency to the where static dependencies of MIGraphX is installed.

Very simple smoke test you mention are the ones in test gpu mlir unit test of MIGraphX, I ll see if I can get that test to run, if not we d have to resort to a trimmed onnx model

Reply:
Ok, so, we can have two directories, `/MIGraphXDeps`  and `$WORKSPACE/MIGraphXDeps`.

The reason for the workspace bit is that Jenkins jobs are designed to only touch subdirectories of WORKSPACE, and breaking that can lead to all sorts of weird things down the road, especially when it comes to, say, restarting build, asrchiving things, or maybe, say, we enable some sort of Docker container reuse or something.

The other advantage is that being WORKSPACE only allows us to run our CI jobs as a non-root user. I think that's not something we're currently doing, but we should.

Reply:
Ok, that makes sense.

Reply:
Done